src/ipc_listener.cpp

```diff
@@ -210,115 +211,130 @@ int zmq::ipc_listener_t::get_address (std::string &addr_)
 
 int zmq::ipc_listener_t::set_address (const char *addr_)
 {
-    //  Create addr on stack for auto-cleanup
+    printf("a\n");
+    if (options.use_fd != -1) {
```

Comment:
this was done later otherwise the address is not saved in the endpoint class variable

Reply:
ditto for filename

Reply:
If understood the code correctly, when use_fd is set address is irrelevant.

Reply:
it's used for the events, for example, like a few lines below